From c21ec52b7491e3891b7bed671346b0186daeac8f Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 20 Jun 2019 21:25:44 -0700
Subject: [PATCH 1/5] cygwin

---
 common.mk | 26 ++++++++++++++++----------
 1 file changed, 16 insertions(+), 10 deletions(-)

diff --git doc-base/common.mk doc-base/common.mk
index a5430cb..73eb5c4 100644
--- doc-base/common.mk
+++ doc-base/common.mk
@@ -8,10 +8,16 @@ override PACKAGE  := doc-base
 
 PATH            := /usr/bin:/usr/sbin:/bin:/sbin:$(PATH)
 
+INSTALL         := install -o $(shell id -u) -g $(shell id -g)
+ifndef _INSTALL
+  # remove --strip-program=true from INSTALL passed by debhelper to
+  # get rid of '--strip-program used without -s option' warnings
+  export _INSTALL := $(subst --strip-program=true,,$(INSTALL))
+endif
 # build abstraction
-install_file    := install -p -o root -g root -m 644
-install_script  := install -p -o root -g root -m 755
-install_dir     := install -d -o root -g root -m 755
+install_file    := $(_INSTALL) -p -m 644
+install_script  := $(_INSTALL) -p -m 755
+install_dir     := $(_INSTALL) -d -m 755
 install_link    := ln -sf
 compress        := gzip -9nf
 
@@ -20,7 +26,7 @@ etcdir          := /etc/$(PACKAGE)
 sbindir         := $(prefix)/sbin
 mandir          := $(prefix)/share/man
 sharedir        := $(prefix)/share/$(PACKAGE)
-perllibdir      := $(prefix)/share/perl5
+perllibdir      := $(prefix)/lib/perl5/site_perl/5.22
 docdir          := $(prefix)/share/doc/$(PACKAGE)
 libdir          := /var/lib/$(PACKAGE)
 omfdir          := $(prefix)/share/omf
@@ -115,15 +121,15 @@ define install
   done;                                                             \
   [ -n "$$files" ] ||                                               \
     echo "$(emptyprefix) installing dir    $(ddirshort)$$dir";      \
-  $(install_dir) "$(DESTDIR)/$$dir";                                \
+  $(install_dir) "$(DESTDIR)//$$dir";                                \
   for file in $$files; do                                           \
       [ -n "$$bfile" ] && tgt="$$dir/$$bfile"  ||                   \
         tgt="$$dir/`basename "$$file"`";                            \
       echo "$(emptyprefix) installing $$what $(ddirshort)$$tgt";    \
-      $$prg "$$file" "$(DESTDIR)/$$tgt";                            \
+      $$prg "$$file" "$(DESTDIR)//$$tgt";                            \
       if [ "$$doCompress" -eq 1 ] ; then                            \
         echo "$(emptyprefix) compressing file  $(ddirshort)$$tgt";  \
-        $(compress) "$(DESTDIR)/$$tgt";                             \
+        $(compress) "$(DESTDIR)//$$tgt";                             \
       fi;                                                           \
   done;
 endef
@@ -133,9 +139,9 @@ define install_links
   set -e;                                                          \
   for file in $2; do                                               \
     echo "$(emptyprefix) installing link   $(ddirshort)$$file";    \
-    $(install_dir) $(DESTDIR)/`dirname "$$file"`;                  \
-    rm -f "$(DESTDIR)/$$file";                                     \
-    $(install_link) "$1" "$(DESTDIR)/$$file";                      \
+    $(install_dir) $(DESTDIR)//`dirname "$$file"`;                  \
+    rm -f "$(DESTDIR)//$$file";                                     \
+    $(install_link) "$1" "$(DESTDIR)//$$file";                      \
   done
 endef
 
-- 
2.20.1


From 27acc1f5cd4caa2479bf8ed609fc084e8277f114 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Fri, 21 Jun 2019 07:34:29 -0700
Subject: [PATCH 2/5] root

---
 perl/Debian/DocBase/DB.pm | 1 -
 1 file changed, 1 deletion(-)

diff --git doc-base/perl/Debian/DocBase/DB.pm doc-base/perl/Debian/DocBase/DB.pm
index 642ab61..85480f7 100644
--- doc-base/perl/Debian/DocBase/DB.pm
+++ doc-base/perl/Debian/DocBase/DB.pm
@@ -74,7 +74,6 @@ sub _SaveDB() { # {{{
   Debug(_g("Saving `%s' (%d)."), $file, $self->{'CHANGED'});
   return unless $self->{'CHANGED'};
   my $readonly = $> != 0;
-  Fatal($ERR_PROCESSING, _g("Needs to be root for this operation.")) if $readonly;
   (my $newfile = $file) =~ s/^[^\.]+/$&-new/g;
   (my $oldfile = $file) =~ s/^[^\.]+/$&-old/g;
 
-- 
2.20.1


From 56fee3de58ada2bb420e0d59f1382eccf6f17fd6 Mon Sep 17 00:00:00 2001
From: furrymcgee <furrymcgee@lippydanger.jumpingcrab.com>
Date: Sun, 28 Jul 2019 00:54:49 +0200
Subject: [PATCH 3/5] warning

---
 perl/Debian/DocBase/DocBaseFile.pm | 5 -----
 1 file changed, 5 deletions(-)

diff --git doc-base/perl/Debian/DocBase/DocBaseFile.pm doc-base/perl/Debian/DocBase/DocBaseFile.pm
index f7dd50f..70e7ef8 100644
--- doc-base/perl/Debian/DocBase/DocBaseFile.pm
+++ doc-base/perl/Debian/DocBase/DocBaseFile.pm
@@ -436,11 +436,6 @@ sub _CheckRequiredFields($$$) { # {{{
       return $self->_PrsErr(PRS_ERR_IGN, _g("format `%s' already defined"), $format);
     }
 
-    if (not grep { $_ eq $format } @SUPPORTED_FORMATS) {
-      $self->_PrsErr(PRS_WARN, _g("format `%s' is not supported"), $$format_data{'format'});
-      next;
-    }
-
     my $index_value = undef;
     # Check `Index' field
     if (grep { $_ eq $format } @NEED_INDEX_FORMATS) {
-- 
2.20.1


From 585ccf27d3d90d7dae321afa1fac296f66e6f33f Mon Sep 17 00:00:00 2001
From: furrymcgee <furrymcgee@lippydanger.jumpingcrab.com>
Date: Tue, 6 Aug 2019 13:10:06 +0200
Subject: [PATCH 4/5] error

---
 perl/Debian/DocBase/DocBaseFile.pm | 1 +
 perl/Debian/DocBase/InstallDocs.pm | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git doc-base/perl/Debian/DocBase/DocBaseFile.pm doc-base/perl/Debian/DocBase/DocBaseFile.pm
index 70e7ef8..bf6f809 100644
--- doc-base/perl/Debian/DocBase/DocBaseFile.pm
+++ doc-base/perl/Debian/DocBase/DocBaseFile.pm
@@ -11,6 +11,7 @@ use warnings;
 
 use File::Glob ':glob';
 use Debian::DocBase::Common;
+use Debian::DocBase::DB;
 use Debian::DocBase::Utils;
 use Debian::DocBase::Gettext;
 use Scalar::Util qw(weaken);
diff --git doc-base/perl/Debian/DocBase/InstallDocs.pm doc-base/perl/Debian/DocBase/InstallDocs.pm
index bbb6e58..076321e 100644
--- doc-base/perl/Debian/DocBase/InstallDocs.pm
+++ doc-base/perl/Debian/DocBase/InstallDocs.pm
@@ -107,7 +107,7 @@ sub _HandleCheck() { # {{{
     my $docfile = Debian::DocBase::DocBaseFile->new($file, 1);
     $docfile->Parse();
     if ($docfile->Invalid()) {
-        Inform(_g("%s: Fatal error found, the file won't be registered."), $file);
+        Error(_g("%s: Fatal error found, the file won't be registered."), $file);
     } elsif ((my $cnt = $docfile->GetWarnErrCount()) > 0) {
         my $msg = _ng("%d warning or non-fatal error found.",
                    "%d warnings or non-fatal errors found.",
-- 
2.20.1


From e4886c35d198eff3b0a3f0eb46cab7192f2f900b Mon Sep 17 00:00:00 2001
From: furrymcgee <furrymcgee@lippydanger.jumpingcrab.com>
Date: Sat, 16 May 2020 20:00:00 +0200
Subject: [PATCH 5/5] xsl:include : unable to load
 /tmp/Y%3A/doc-base/doc/xsl/common.xsl

---
 doc/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git doc-base/doc/Makefile doc-base/doc/Makefile
index 78a7353..213c111 100644
--- doc-base/doc/Makefile
+++ doc-base/doc/Makefile
@@ -7,6 +7,7 @@ ALL_TARGET := build-local
 SUBDIRS		 := 
 include ../common.mk
 
+CURDIR   := ..
 XMLLINT  := xmllint --nonet --noout --postvalid --xinclude
 XSLTPROC := xsltproc --nonet --xinclude
 
-- 
2.20.1

