From 80e9b7527a22316cf3251fb9b8ae8984e64a426f Mon Sep 17 00:00:00 2001
From: furrymcgee <furrymcgee@lippydanger.jumpingcrab.com>
Date: Sun, 23 May 2021 20:00:00 +0200
Subject: [PATCH] index on (no branch): d326a3185 Version v0.7.8

---
 src/utils/path.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git vifm/src/utils/path.c vifm/src/utils/path.c
index 34b2d9876..35070884e 100644
--- vifm/src/utils/path.c
+++ vifm/src/utils/path.c
@@ -302,7 +302,7 @@ is_root_dir(const char *path)
 int
 is_unc_root(const char *path)
 {
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
 	if(is_unc_path(path) && path[2] != '\0')
 	{
 		char *p = strchr(path + 2, '/');
@@ -692,7 +692,7 @@ find_cmd_in_path(const char cmd[], size_t path_len, char path[])
 	return 1;
 }
 
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
 
 int
 is_unc_path(const char path[])
-- 
2.31.1

