From dd49e04d1e9782de8605efca9cff47b7427292d8 Mon Sep 17 00:00:00 2001
From: furrymcgee <furrymcgee@lippydanger.jumpingcrab.com>
Date: Thu, 15 Aug 2019 11:38:16 +0200
Subject: [PATCH 1/2] e6b3ba4 dh_testroot: root_requirements() no longer read
 d/control

---
 debian/rules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git dh-autoreconf/debian/rules dh-autoreconf/debian/rules
index e020f7f..1304204 100755
--- dh-autoreconf/debian/rules
+++ dh-autoreconf/debian/rules
@@ -3,7 +3,7 @@ include /usr/share/dpkg/pkg-info.mk
 pod2man := pod2man -r "dh-autoreconf v$(DEB_VERSION)" -c dh-autoreconf
 
 %:
-	dh $@
+	DEB_RULES_REQUIRES_ROOT=no dh $@ --without autoreconf
 
 override_dh_auto_build:
 	$(pod2man) --section=1 dh_autoreconf dh_autoreconf.1
-- 
2.20.1


From 54dcdbba852b9f4830b69044f303ae09af7c13be Mon Sep 17 00:00:00 2001
From: furrymcgee <furrymcgee@lippydanger.jumpingcrab.com>
Date: Thu, 15 Aug 2019 16:57:23 +0200
Subject: [PATCH 2/2] Can't exec "dh_strip_nondeterminism": No such file or
 directory

Disable as they are unneeded (and we can then be sure debhelper builds without needing autotools-dev, dh-strip-nondetermism etc.)
---
 debian/rules | 4 ++++
 1 file changed, 4 insertions(+)

diff --git dh-autoreconf/debian/rules dh-autoreconf/debian/rules
index 1304204..d6b345c 100755
--- dh-autoreconf/debian/rules
+++ dh-autoreconf/debian/rules
@@ -5,6 +5,10 @@ pod2man := pod2man -r "dh-autoreconf v$(DEB_VERSION)" -c dh-autoreconf
 %:
 	DEB_RULES_REQUIRES_ROOT=no dh $@ --without autoreconf
 
+override_dh_strip_nondeterminism:
+
+override_dh_installman:
+
 override_dh_auto_build:
 	$(pod2man) --section=1 dh_autoreconf dh_autoreconf.1
 	$(pod2man) --section=1 dh_autoreconf_clean dh_autoreconf_clean.1
-- 
2.20.1

